src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1198,68 +1379,70 @@ static bool addTriggeredSubscriptions_noCache
-   * http://stackoverflow.com/questions/15966991/mongodb-reverse-regex/15989520).
-   * Note that although we are using a isPattern=true in the MongoDB query besides $where, we
-   * also need to check that in the if statement in the JavaScript function given that a given
-   * sub document could include both isPattern=true and isPattern=false documents */
-  std::string function = std::string("function()") +
-         "{" +
-            "for (var i=0; i < this."+CSUB_ENTITIES+".length; i++) {" +
-                "if (this."+CSUB_ENTITIES+"[i]."+CSUB_ENTITY_ISPATTERN+" == \"true\" && " +
-                    "(this."+CSUB_ENTITIES+"[i]."+CSUB_ENTITY_TYPE+" == \""+entityType+"\" || " +
-                        "this."+CSUB_ENTITIES+"[i]."+CSUB_ENTITY_TYPE+" == \"\" || " +
-                        "!(\""+CSUB_ENTITY_TYPE+"\" in this."+CSUB_ENTITIES+"[i])) && " +
-                    "\""+entityId+"\".match(this."+CSUB_ENTITIES+"[i]."+CSUB_ENTITY_ID+")) {" +
-                    "return true; " +
-                "}" +
-            "}" +
-            "return false; " +
-         "}";
-  LM_T(LmtMongo, ("JS function: %s", function.c_str()));
+  std::string entIdQ        = CSUB_ENTITIES   "." CSUB_ENTITY_ID;
+  std::string entTypeQ      = CSUB_ENTITIES   "." CSUB_ENTITY_TYPE;
+  std::string entPatternQ   = CSUB_ENTITIES   "." CSUB_ENTITY_ISPATTERN;
+  std::string typePatternQ  = CSUB_ENTITIES   "." CSUB_ENTITY_ISTYPEPATTERN;
+  std::string inRegex       = "{ $in: [ " + spathRegex + ", null ] }";
+  BSONObj     spBson        = fromjson(inRegex);
+
+  /* Query is an $or of 4 sub-clauses:
+   *
+   * idNPtypeNP -> id no pattern, type no pattern
+   * idPtypeNP  -> id pattern,    type no pattern
+   * idNPtypeP  -> id no pattern, type pattern
+   * idPtypeP   -> id pattern,    type pattern
+   *
+   * The last three use $where to search for patterns in DB. As far as I know, this is the only
+   * way to do a "reverse regex" query in MongoDB (see http://stackoverflow.com/questions/15966991/mongodb-reverse-regex/15989520).
+   * The first part of the "if" in these function is used to ensure that the function matches the corresponding
+   * pattern/no-pattern combination, as the entity vector may contain a mix.
+   *
+   * Note that we are using the construct:
+   *
+   *   typePatternQ << BSON("$ne" << true)
+   *
+   * instead of just
+   *
+   *   typePatternQ << false
+   *
+   * as the former also matches documents without the typePatternQ (i.e. legacy sub documents created before the
+   * isTypePattern feature was developed)
+   *
+   * FIXME: condTypeQ, condValueQ and servicePath part could be "factorized" out of the $or clause
+   */
 
-  BSONObjBuilder  queryPattern;
+  //
+  // Allocating buffer to hold all these BIG variables, necessary for the population of 
+  // the four parts of the final query.
+  // The necessary variables are too big for the stack and thus moved to the head, inside BsonGroup.
+  //
+  BsonGroup* bgP = new BsonGroup();
```

Comment:
I looked at it, in the entire file. With the diffs it's not possible to see anything.

It looks good to me.